internal/webserver/wsauthconfig/config.go

```diff
@@ -0,0 +1,46 @@
+package wsauthconfig
+
+type OAuth2Conf struct {
+	ClientID     string   `validate:"required"` // OAuth client id (this app)
+	ClientSecret string   // OAuth2 secret (associated with ClientID, optional)
+	RedirectURL  string   `validate:"required"` // where should the OAuth2 provider return the user to
+	Scopes       []string // list of scopes to ask for from the OAuth2 provider
+}
+
+type OIDCConf struct {
+	IssuerURL string `validate:"url,omitempty"`
+
+	// the ones below are only needed if the OIDC discovery mechanism doesn't work
+	AuthURL     string `validate:"omitempty,url"`
+	TokenURL    string `validate:"omitempty,url"`
+	UserInfoURL string `validate:"omitempty,url"`
+	Algorithms  []string
+}
+
+// config for JWT token signature check
+type JWTConf struct {
+	UseJWKS bool `bool:"UseJWKS" validate:"required"`
+	// used when UseJWKS is set to true
+	JwksURL              string   `validate:"required_if=UseJWKS true,url,omitempty"`
+	JwksSignatureMethods []string `validate:"required_if=UseJWKS true,omitempty"`
+	// used when UseJWKS is set to false
```

Comment:
with the validation the comments seem a bit superfluous

Reply:
I'll clean this up later